jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -0,0 +1,70 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import org.eclipse.jetty.http.pathmap.MatchedResource;
+import org.eclipse.jetty.http.pathmap.PathMappings;
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Handler that delegates to other handlers through a configured {@link PathMappings}.
+ */
+public class PathMappingsHandler extends Handler.Wrapper
+{
+    private static final Logger LOG = LoggerFactory.getLogger(PathMappingsHandler.class);
+
+    private final PathMappings<Handler> mappings = new PathMappings<>();
+
+    public void addMapping(PathSpec pathSpec, Handler handler)
+    {
+        if (isStarted())
+            throw new IllegalStateException("Cannot add mapping: " + this);
+
```

Comment:
maybe some loop detection would be good?

Reply:
What kind of loop detection would fit here?
Got a scenario?

Or do you mean a Handler from earlier in the tree being mapped under here and possibly being a loop?
```
 server.handler
   MyHandler@0xBEEF
      PathMappingHandler
         mapping("/*", MyHandler@0xBEEF)
```

If that's the case, do we have a generic version of this kind of loop detection?
As it could be useful in many scenarios (even with just the existing Handler.Collection / Handler.Wrapper impls)

Reply:
Done